addons/account/models/account_order.py

```diff
@@ -0,0 +1,229 @@
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import Command, _, api, fields, models
+from odoo.tools import float_is_zero
+
+
+class AccountOrder(models.AbstractModel):
+    # This class enables downpayment support for models that can have down payments on them
+    _name = 'account.order'
+    _description = 'Account Order'
+
+    amount_to_invoice = fields.Monetary("Amount to invoice", store=True, compute='_compute_amount_to_invoice')
+    amount_invoiced = fields.Monetary("Already invoiced", compute='_compute_amount_invoiced')
+
+    # To override
+    company_id = fields.Many2one('res.company')
+    currency_id = fields.Many2one('res.currency')
+    fiscal_position_id = fields.Many2one('account.fiscal.position', string='Fiscal Position')
+    partner_id = fields.Many2one('res.partner')
+
+    amount_total = fields.Monetary()
+
+    invoice_ids = fields.Many2many('account.move')
+    order_line = fields.One2many('account.order.line')
```

Comment:
```suggestion
    order_line_id = fields.One2many('account.order.line')
```

Reply:
![image](https://github.com/odoo/odoo/assets/3251665/3c87995e-43d8-4ef0-b331-b303a26c0e38)

Reply:
I doubt all these are correct matches of the field, and you already did rename `invoice_lines` to `invoice_line_ids` so why not... I don't really care but I'll let the final reviewer decide